PsfRuntime/ArgRedirection.h

```diff
@@ -0,0 +1,186 @@
+    }
+
+    size_t remBufSize = MAX_CMDLINE_PATH - strlenImpl(cnvtCmdLine);
+    if (isUnderAppDataRoaming)
+    {
+        std::filesystem::path roamingAppData = psf::known_folder(FOLDERID_RoamingAppData);
+        auto packageRoamingAppDataPath = psf::known_folder(FOLDERID_LocalAppData) / std::filesystem::path(L"Packages") / psf::current_package_family_name() / LR"(LocalCache\Roaming)";
+        std::wstring relativePath = widen(fileName + (roamingAppData.native().length()));
+        std::wstring fullPath = packageRoamingAppDataPath.native() + relativePath;
+
+        if (GetFileAttributesImpl(fullPath.c_str()) != INVALID_FILE_ATTRIBUTES)
+        {
+            result = true;
+            strcatImpl(cnvtCmdLine, remBufSize, (CharT*)(packageRoamingAppDataPath.c_str()));
+            strcatImpl(cnvtCmdLine, remBufSize, fileName + (roamingAppData.native().length()));
+        }
+    }
+    return result;
+}
+
+template <typename CharT>
+bool IsUnderUserAppDataLocalandReplace(const CharT* fileName, CharT* cnvtCmdLine)
+{
+    bool isUnderAppDataLocal = false;
+    bool result = false;
+
+    if (fileName == NULL)
+    {
+        return result;
+    }
+    constexpr wchar_t root_local_device_prefix[] = LR"(\\?\)";
+    constexpr wchar_t root_local_device_prefix_dot[] = LR"(\\.\)";
+
+    if (std::equal(root_local_device_prefix, root_local_device_prefix + 4, fileName))
+    {
+        isUnderAppDataLocal = is_path_relative(fileName + 4, psf::known_folder(FOLDERID_LocalAppData));
+    }
+    else if (std::equal(root_local_device_prefix_dot, root_local_device_prefix_dot + 4, fileName))
+    {
+        isUnderAppDataLocal = is_path_relative(fileName + 4, psf::known_folder(FOLDERID_LocalAppData));
+    }
+    else
+    {
+        isUnderAppDataLocal = is_path_relative(fileName, psf::known_folder(FOLDERID_LocalAppData));
+    }
+
+    size_t remBufSize = MAX_CMDLINE_PATH - strlenImpl(cnvtCmdLine);
+    if (isUnderAppDataLocal)
+    {
+        std::filesystem::path localAppdata = psf::known_folder(FOLDERID_LocalAppData);
+        auto packageLocalAppDataPath = psf::known_folder(FOLDERID_LocalAppData) / std::filesystem::path(L"Packages") / psf::current_package_family_name() / LR"(LocalCache\Local)";
+        std::wstring relativePath = widen(fileName + (localAppdata.native().length()));
+        std::wstring fullPath = packageLocalAppDataPath.native() + relativePath;
+
+        if (GetFileAttributesImpl(fullPath.c_str()) != INVALID_FILE_ATTRIBUTES)
+        {
+            result = true;
+            strcatImpl(cnvtCmdLine, remBufSize, (CharT*)(packageLocalAppDataPath.c_str()));
+            strcatImpl(cnvtCmdLine, remBufSize, fileName + (localAppdata.native().length()));
+        }
+    }
+    return result;
+}
+
+template <typename CharT>
+void convertCmdLineParameters(CharT* inpCmdLine, CharT* cnvtCmdLine)
+{
+    CharT* next_token;
+    size_t cmdLinelen = strlenImpl(inpCmdLine);
+    CharT* cmdLine = new CharT[cmdLinelen+1];
```

Comment:
Please wrap this into a std::unique_ptr<CharT>.  This avoids us having to manually delete the array,